src/hardware/input/mouse_config.cpp

```diff
@@ -399,18 +472,41 @@ static void config_init(Section_prop& secprop)
 	        "      Windows 9x under DOSBox. Under Windows 3.x, the driver is not disabled,\n"
 	        "      but the Windows 3.x mouse driver takes over.");
 
-	prop_bool = secprop.Add_bool("dos_mouse_immediate", always, false);
-	assert(prop_bool);
-	prop_bool->Set_help(
-	        "Update mouse movement counters immediately, without waiting for interrupt\n"
-	        "('off' by default). May improve gameplay, especially in fast-paced games\n"
-	        "(arcade, FPS, etc.), as for some games it effectively boosts the mouse\n"
-	        "sampling rate to 1000 Hz, without increasing interrupt overhead.\n"
-	        "Might cause compatibility issues. List of known incompatible games:\n"
-	        "  - Ultima Underworld: The Stygian Abyss\n"
-	        "  - Ultima Underworld II: Labyrinth of Worlds\n"
-	        "Please report it if you find another incompatible game so we can update this\n"
-	        "list.");
+	prop_bool = secprop.Add_bool("dos_mouse_driver", deprecated, true);
+	prop_bool->Set_help("Renamed to 'builtin_dos_mouse_driver'.");
+
+	prop_str = secprop.Add_string("builtin_dos_mouse_driver_model",
+	                              always,
+	                              model_dos_wheel_str);
+	assert(prop_str);
+	prop_str->Set_values(
+	        {model_dos_2button_str, model_dos_3button_str, model_dos_wheel_str});
+	prop_str->Set_help(
+	        "Set the mouse model to be simulated by the built-in DOS mouse driver ('wheel'\n"
+	        "by default).\n"
+	        "  2button:  2 buttons, use for some old games which malfunction if the middle\n"
+	        "            button gets pressed.\n"
+	        "  3button:  3 buttons, only supported by very few games.\n"
+	        "  wheel:    3 buttons + wheel, supports the CuteMouse WheelAPI version 1.0.");
+
+	prop_str = secprop.Add_string("builtin_dos_mouse_driver_options", always, "");
+	assert(prop_str);
+	prop_str->Set_help(
+	        "Additional built-in DOS mouse driver settings, space or comma separated\n"
+	        "(unset by default).\n"
+	        "  immediate:  Update mouse movement counters immediately, without waiting for\n"
+	        "              interrupt. May improve mouse movement reaction time in fast-paced\n"
+	        "              games (arcade, FPS, etc.), but causes issues in some titles.\n"
+	        "              List of known incompatible games:\n"
+	        "                - Ultima Underworld: The Stygian Abyss\n"
+	        "                - Ultima Underworld II: Labyrinth of Worlds\n"
+	        "              Please report it if you find another incompatible game so we can\n"
+	        "              update this list.\n"
+	        "  modern:     Enable modern (v7.0+) mouse driver behavior. Needed at least by\n"
+	        "              Descent II with the official Voodoo patch.");
```

Comment:
I see you've shoehorned this in 😆 I was on the fence with this because it's a bit of a stretch to include a config option for a single game revision that's not even playable, lol, but perhaps it's useful for other very late DOS era games. We don't wanna our config to look like DOSBox-X's where you need to be a hardware expert to configure anything 😅 

Maybe this would be a bit more descriptive:

If provided, v7.0+ Microsoft mouse driver behaviour is emulated, otherwise the v6.0 and earlier behaviour (the two are slightly incompatible). Only Descent II with the official Voodoo patch has been found to require the v7.0+ behaviour so far.

Reply:
OK, changed the description.